docs/how_to_guides/hosting_validator_models.md

```diff
@@ -0,0 +1,210 @@
+# Hosting Validator Models
+
+Validation using machine learning models is a highly effective way of detecting LLM hallucinations that are not easily or possibly detected using traditional coding techniques. We’ve selected and tuned some ML models to run different validations. These models are wrapped within model validators. This makes the access pattern straightforward — you can use model-based validators the same as any other [validator](https://www.guardrailsai.com/docs/concepts/validators).
+
+The default access pattern is to use these validators locally for development, but this causes a few issues in production. These models have a large memory footprint, and if they’re run in on-process environments without the correct infrastructure (GPUs, memory, etc.), they can run slowly and suboptimally, adding a huge amount of latency to your application.
+
+In this guide, we will walk through hosting a Guardrails validation model with FastAPI.
+
+## Step 1: Pick a Validator
+
+First, choose a validator that you want to use. You can browse a list of all the Guardrails validators on [Guardrails Hub](https://hub.guardrailsai.com/).
+
+:::note
+We suggest picking the “ML” filter on the left side of the page in the “Infrastructure Requirements” section to find validators that use models.
+:::
+
+![alt_text](./assets/infrastructure_requirements.png "image_tooltip")
+
+For the purpose of this guide, let’s use the [ToxicLanguage validator](https://hub.guardrailsai.com/validator/guardrails/toxic_language).
+
+## Step 2: Find and Download the Validator Model
+
+The validator model is referenced in the validator implementation, as well as in the README.
+
+![alt_text](./assets/validator_in_readme.png "image_tooltip")
+
+As we can see, the ToxicLanguage validator currently uses the [Detoxify model](https://github.com/unitaryai/detoxify). In order to use this, we need to install the necessary libraries:
+
+```bash
+# Install FastAPI server dependencies
+pip install fastapi uvicorn
+
+# Option 1 - Install required validator dependencies(General approach - Requires Guardrails)
+guardrails hub install hub://guardrails/toxic_language --install-local-models
+
+# Option 2 - Install only required dependencies (Validator Specific - From README)
+pip install detoxify torch
```

Comment:
remove this option, mention it as part of #4 here https://github.com/guardrails-ai/guardrails/pull/1008/files#r1717344125